R/user_guide.R

```diff
@@ -281,60 +272,209 @@ render_setup_instructions <- function(path, pr) {
+            tags$div(
+              tags$span(class="emphasized", "Username:"),
+              "rstudio-connect"
+            ),
+            tags$div(
+              tags$span(class="emphasized", "Password:"),
+              tags$span(class="italic", "any valid API key from RStudio Connect")
+            )
+          ),
+          tags$li("Select whether to Require SSL"),
+          tags$li("Save changes")
+        )
+      )
+    )
+  }
+
+  as.character(htmltools::htmlTemplate(
+    system.file("template/index.html", package = "plumbertableau", mustWork = TRUE),
+    title_desc = title_desc,
+    body_content = body_content
+  ))
+}
+
+#' @importFrom htmltools tags
+create_help <- function(pr) {
+  function(req, res) {
+    "!DEBUG `write_log_message(req, res, 'Generating Tableau Analytical Extension Overview')"
+    # Parse the endpoint path from header on RStudio Connect
+    content_path <- NULL
+    if (!rlang::is_null(req$HTTP_RSTUDIO_CONNECT_APP_BASE_URL)) {
+      base_url <- req$HTTP_RSTUDIO_CONNECT_APP_BASE_URL
+      rsc_root <- Sys.getenv("CONNECT_SERVER")
+      content_path <- gsub(rsc_root,
+                           "",
+                           base_url,
+                           fixed = TRUE
+      )
+
+      # If the path requested is not root (/), strip it from the content path
+      if (req$PATH_INFO != "/") {
+        content_path <- gsub(req$PATH_INFO,
+                             "",
+                             content_path,
+                             fixed = TRUE
         )
+      }
+
+      # Ensure path starts with '/'
+      if (!stringi::stri_startswith(content_path, fixed = "/")) content_path <- paste0("/", content_path)
+    }
+    render_help(content_path, pr)
+  }
+}
+
+render_help <- function(path, pr) {
+  apiSpec <- pr$getApiSpec()
+  title <- apiSpec$info$title
+  version <- apiSpec$info$version
+  desc <- markdown::markdownToHTML(text = apiSpec$info$user_description,
+                                    fragment.only = TRUE)
+
+  title_desc <- htmltools::tagList(
+    tags$h1(
+      class="padded-fully title",
+      title,
+      if (!is.null(version)) paste0("(v", version, ")")
+    ),
+    tags$div(class = "api-desc",
+      tags$div(
+        class="padded-flat-top",
+        htmltools::HTML(desc)
       )
     )
   )
 
+  body_content <- htmltools::tagList(
+    tags$h3(
+      class="subtitle",
+      "Read up on plumbertableau, Tableau, and RStudio Connect"
+    ),
+    tags$div(
+      class = "padded-fully",
+      htmltools::HTML(markdown::markdownToHTML(
+        text = help_text()), fragment.only = TRUE
+      )
+    )
+  )
+  
   as.character(htmltools::htmlTemplate(
     system.file("template/index.html", package = "plumbertableau", mustWork = TRUE),
-    content = ui
+    title_desc = title_desc,
+    body_content = body_content
   ))
 }
 
-strip_md_links <- function(text) {
-  stringi::stri_replace_all(text, regex = " +\\*.*", "")
+help_text <- function() {
```

Comment:
Agree with @toph-allen that we'll want this to come directly from the vignette if possible, but that can be explored in a separate PR